addons/stock/models/stock_move_line.py

```diff
@@ -199,6 +199,12 @@ def create(self, vals_list):
         mls = super(StockMoveLine, self).create(vals_list)
 
         for ml in mls:
+            if ml.move_id and \
+                    ml.move_id.picking_id and \
+                    ml.move_id.picking_id.immediate_transfer and \
+                    ml.move_id.state != 'done' and \
+                    'qty_done' in vals:
+                ml.move_id.product_uom_qty = ml.move_id.quantity_done
```

Comment:
move_id.product_uom_qty will be write multi time.

replace by:
```python
if 'qty_done' in vals:
     moves = self.filtered(lambda ml: ml.move_id.state != 'done' and ml.move_id.picking_id and ml.move_id.picking_id.immediate_transfer).mapped('move_id')
     for move in moves:
          move.product_uom_qty = move.quantity_done
```

Reply:
first condition has 'ml.move_id.state != 'done' and second ml.state == 'done' which is complementary

Reply:
but if there are multi ml per move you write the same information multi time.